api/modelgeneration/modelgeneration.go

```diff
@@ -27,10 +26,9 @@ func NewClient(st base.APICallCloser) *Client {
 }
 
 // AddGeneration adds a model generation to the config.
-func (c *Client) AddGeneration(model names.ModelTag) error {
```

Comment:
I made a comment on this in the other PR.

Basically it makes the client usage a bit more terse and cogent - caller passes the model ID to indicate a model, without having to know about any coercion necessary for the server call. And said coercion can happen pre-flight in one place rather than at all sites where the client is used.

I have previously reverted changes that passed tags around (albeit on the server side). I was lead to believe they were an in-flight way of passing typed IDs in the generic `Entities` DTOs.